apps/alert_processor/lib/service_info/service_info_cache.ex

```diff
@@ -652,14 +658,31 @@ defmodule AlertProcessor.ServiceInfoCache do
   end
 
   defp fetch_route_branches("Red") do
-    {:ok, route_shapes} = ApiClient.route_shapes("Red")
+    {:ok, patterns, included} = ApiClient.route_patterns_with_stops("Red")
 
-    route_shapes
-    |> Enum.reject(fn %{"attributes" => %{"priority" => priority}} ->
-      priority == -1
+    included_by_type =
+      included
+      |> Enum.group_by(fn %{"type" => ty} -> ty end)
```

Comment:
suggestion: Stylistically it's more conventional to only use pipelines when more than one pipe operator is involved; otherwise you could just call the function "normally" and save some space. So (with some additional destructuring and use of `&` to reduce variable duplication) this could be condensed to:

```elixir
%{"stop" => stops, "trip" => trips} = Enum.group_by(included, & &1["type"])
stops_by_id = Map.new(stops, &{&1["id"], &1})
trips_by_id = Map.new(trips, &{&1["id"], &1})
```

Or, with more up-front failures, since I did just write that comment about using `fetch!`:

```elixir
%{"stop" => stops, "trip" => trips} = Enum.group_by(included, &Map.fetch!(&1, "type"))
stops_by_id = Map.new(stops, fn %{"id" => id} = stop -> {id, stop} end)
trips_by_id = Map.new(trips, fn %{"id" => id} = trip -> {id, trip} end)
```